### ISSUE TITLE
Package installs failing on Windows with conda 4.3.x due to cryptography error

### ISSUE THREAD

**Issue by kmdouglass** [conda/conda#4429]
Hi all,
I am unable to install my packages from my Anaconda Cloud channel with conda in Windows with conda 4.3.x.

The command `conda create -n bstore -c kmdouglass bstore` results in the following error repeated to the screen a few hundred times:

`extern "Python": function Cryptography_locking_cb() called, but @ffi.def_extern() was not called in the current subinterpreter. Returning 0.`

The final result is that the environment is created but no packages are installed into it. Attempts at manually installing packages one-by-one result in errors stating that the package is not available for win-64, even though they are available. Using conda 4.2.16 instead, everything goes smoothly.

The same conda command in Linux x64 and conda 4.3.8 works perfectly.

I think the issue may be related to this one, but I am not sure: https://github.com/pyca/cryptography/issues/3282

Anyone have any ideas if this is fixable, or should I wait until a new version of conda is pushed out?

-----

## Working conda environment

```

(C:\Users\laboleb\Anaconda3) C:\Users\laboleb>conda info
Current conda install:

               platform : win-64
          conda version : 4.2.16
       conda is private : False
      conda-env version : 4.2.16
    conda-build version : 2.1.1
         python version : 3.5.1.final.0
       requests version : 2.12.4
       root environment : C:\Users\laboleb\Anaconda3  (writable)
    default environment : C:\Users\laboleb\Anaconda3
       envs directories : C:\Users\laboleb\Anaconda3\envs
          package cache : C:\Users\laboleb\Anaconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
                          https://conda.anaconda.org/soft-matter/win-64
                          https://conda.anaconda.org/soft-matter/noarch
            config file : C:\Users\laboleb\.condarc
           offline mode : False
```

## Not working conda environment

```
(C:\Users\laboleb\Anaconda3) C:\Users\laboleb>conda info
Current conda install:

               platform : win-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : 2.1.1
         python version : 3.5.2.final.0
       requests version : 2.12.4
       root environment : C:\Users\laboleb\Anaconda3  (writable)
    default environment : C:\Users\laboleb\Anaconda3
       envs directories : C:\Users\laboleb\Anaconda3\envs
          package cache : C:\Users\laboleb\Anaconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
                          https://conda.anaconda.org/soft-matter/win-64
                          https://conda.anaconda.org/soft-matter/noarch
            config file : C:\Users\laboleb\.condarc
           offline mode : False
             user-agent : conda/4.3.8 requests/2.12.4 CPython/3.5.2 Windows/8.1
Windows/6.3.9600
```

**Comment by kalefranz** [comment]
Thanks for the detailed report.

We've had so many problems with pyopenssl and cryptography lately.  It's unfortunate.

Can you give me the `conda list` output for both the 4.2.16 and 4.3.8 conda environments?  I don't think there are any logic errors in conda's own code.  It's probably a dependency mis-match between requests, pyopenssl, and all the related sub-dependencies.  We just need to get the correct version constraints set, which is done for each package.

**Comment by kmdouglass** [comment]
Thanks for the fast reply. I should add that this behavior is reproducible for me across two different Windows machines. I am also not sure that pyopenssl/cryptography are the real causes of the problem, but it seems like a good place to start looking.

In the previous message, I took `conda info` output from two different but equally affected Windows machines. For better consistency, I've included `conda info` from the same machine in working/not working conditions below.

# Working environment
<details>
<summary>conda list</summary>
# packages in environment at C:\Users\laboleb\Anaconda3:
#
_nb_ext_conf              0.2.0                    py35_0  
alabaster                 0.7.8                    py35_0  
anaconda                  4.1.0               np111py35_0  
anaconda-client           1.4.0                    py35_0  
anaconda-navigator        1.2.1                    py35_0  
argcomplete               1.0.0                    py35_1  
astropy                   1.2.1               np111py35_0  
babel                     2.3.3                    py35_0  
backports                 1.0                      py35_0  
beautifulsoup4            4.4.1                    py35_0  
bitarray                  0.8.1                    py35_1  
bokeh                     0.11.1                   py35_0  
boto                      2.40.0                   py35_0  
bottleneck                1.0.0               np111py35_1  
bzip2                     1.0.6                    vc14_3  [vc14]
cffi                      1.6.0                    py35_0  
chardet                   2.3.0                    py35_0  
chest                     0.2.3                    py35_0  
click                     6.6                      py35_0  
cloudpickle               0.2.1                    py35_0  
clyent                    1.2.2                    py35_0  
colorama                  0.3.7                    py35_0  
comtypes                  1.1.2                    py35_0  
conda                     4.2.16                   py35_0  
conda-build               2.1.1                    py35_0  
conda-env                 2.6.0                         0  
conda-verify              2.0.0                    py35_0  
configobj                 5.0.6                    py35_0  
console_shortcut          0.1.1                    py35_1  
contextlib2               0.5.3                    py35_0  
cryptography              1.4                      py35_0  
curl                      7.49.0                   vc14_0  [vc14]
cycler                    0.10.0                   py35_0  
cython                    0.24                     py35_0  
cytoolz                   0.8.0                    py35_0  
dask                      0.10.0                   py35_0  
datashape                 0.5.2                    py35_0  
decorator                 4.0.10                   py35_0  
dill                      0.2.5                    py35_0  
docutils                  0.12                     py35_2  
dynd-python               0.7.2                    py35_0  
entrypoints               0.2.2                    py35_0  
et_xmlfile                1.0.1                    py35_0  
fastcache                 1.0.2                    py35_1  
filelock                  2.0.7                    py35_0  
flask                     0.11.1                   py35_0  
flask-cors                2.1.2                    py35_0  
freetype                  2.5.5                    vc14_1  [vc14]
get_terminal_size         1.0.0                    py35_0  
gevent                    1.1.1                    py35_0  
greenlet                  0.4.10                   py35_0  
h5py                      2.6.0               np111py35_0  
hdf5                      1.8.15.1                 vc14_4  [vc14]
heapdict                  1.0.0                    py35_1  
idna                      2.1                      py35_0  
imagesize                 0.7.1                    py35_0  
ipykernel                 4.3.1                    py35_0  
ipython                   4.2.0                    py35_0  
ipython_genutils          0.1.0                    py35_0  
ipywidgets                4.1.1                    py35_0  
itsdangerous              0.24                     py35_0  
jdcal                     1.2                      py35_1  
jedi                      0.9.0                    py35_1  
jinja2                    2.8                      py35_1  
jpeg                      8d                       vc14_0  [vc14]
jsonschema                2.5.1                    py35_0  
jupyter                   1.0.0                    py35_3  
jupyter_client            4.3.0                    py35_0  
jupyter_console           4.1.1                    py35_0  
jupyter_core              4.1.0                    py35_0  
libdynd                   0.7.2                         0  
libpng                    1.6.22                   vc14_0  [vc14]
libtiff                   4.0.6                    vc14_2  [vc14]
llvmlite                  0.11.0                   py35_0  
locket                    0.2.0                    py35_1  
lxml                      3.6.0                    py35_0  
markupsafe                0.23                     py35_2  
matplotlib                1.5.1               np111py35_0  
menuinst                  1.4.1                    py35_0  
mistune                   0.7.2                    py35_0  
mkl                       11.3.3                        1  
mkl-service               1.1.2                    py35_2  
mpmath                    0.19                     py35_1  
multipledispatch          0.4.8                    py35_0  
nb_anacondacloud          1.1.0                    py35_0  
nb_conda                  1.1.0                    py35_0  
nb_conda_kernels          1.0.3                    py35_0  
nbconvert                 4.2.0                    py35_0  
nbformat                  4.0.1                    py35_0  
nbpresent                 3.0.2                    py35_0  
networkx                  1.11                     py35_0  
nltk                      3.2.1                    py35_0  
nose                      1.3.7                    py35_1  
notebook                  4.2.1                    py35_0  
numba                     0.26.0              np111py35_0  
numexpr                   2.6.0               np111py35_0  
numpy                     1.11.0                   py35_2  
odo                       0.5.0                    py35_0  
openpyxl                  2.3.2                    py35_0  
openssl                   1.0.2h                   vc14_0  [vc14]
pandas                    0.18.1              np111py35_0  
partd                     0.3.4                    py35_0  
path.py                   8.2.1                    py35_0  
pathlib2                  2.1.0                    py35_0  
patsy                     0.4.1                    py35_0  
pep8                      1.7.0                    py35_0  
pickleshare               0.7.2                    py35_0  
pillow                    3.2.0                    py35_1  
pip                       8.1.2                    py35_0  
pkginfo                   1.4.1                    py35_0  
ply                       3.8                      py35_0  
psutil                    4.3.0                    py35_0  
py                        1.4.31                   py35_0  
pyasn1                    0.1.9                    py35_0  
pycosat                   0.6.1                    py35_1  
pycparser                 2.14                     py35_1  
pycrypto                  2.6.1                    py35_4  
pycurl                    7.43.0                   py35_0  
pyflakes                  1.2.3                    py35_0  
pygments                  2.1.3                    py35_0  
pyopenssl                 16.2.0                   py35_0  
pyparsing                 2.1.4                    py35_0  
pyqt                      4.11.4                   py35_6  
pyreadline                2.1                      py35_0  
pytables                  3.2.2               np111py35_4  
pytest                    2.9.2                    py35_0  
python                    3.5.1                         5  
python-dateutil           2.5.3                    py35_0  
pytz                      2016.4                   py35_0  
pywin32                   220                      py35_1  
pyyaml                    3.11                     py35_4  
pyzmq                     15.2.0                   py35_0  
qt                        4.8.7                    vc14_8  [vc14]
qtconsole                 4.2.1                    py35_0  
qtpy                      1.0.2                    py35_0  
requests                  2.12.4                   py35_0  
rope                      0.9.4                    py35_1  
ruamel_yaml               0.11.14                  py35_0  
scikit-image              0.12.3              np111py35_1  
scikit-learn              0.17.1              np111py35_1  
scipy                     0.17.1              np111py35_1  
setuptools                23.0.0                   py35_0  
simplegeneric             0.8.1                    py35_1  
singledispatch            3.4.0.3                  py35_0  
sip                       4.16.9                   py35_2  
six                       1.10.0                   py35_0  
snowballstemmer           1.2.1                    py35_0  
sockjs-tornado            1.0.3                    py35_0  
sphinx                    1.4.1                    py35_0  
sphinx_rtd_theme          0.1.9                    py35_0  
spyder                    2.3.9                    py35_0  
sqlalchemy                1.0.13                   py35_0  
statsmodels               0.6.1               np111py35_1  
sympy                     1.0                      py35_0  
tk                        8.5.18                   vc14_0  [vc14]
toolz                     0.8.0                    py35_0  
tornado                   4.3                      py35_1  
traitlets                 4.2.1                    py35_0  
unicodecsv                0.14.1                   py35_0  
vs2015_runtime            14.0.25123                    0  
werkzeug                  0.11.10                  py35_0  
wheel                     0.29.0                   py35_0  
xlrd                      1.0.0                    py35_0  
xlsxwriter                0.9.2                    py35_0  
xlwings                   0.7.2                    py35_0  
xlwt                      1.1.2                    py35_0  
zlib                      1.2.8                    vc14_3  [vc14]
</details>

<details>
<summary>conda info</summary>
Current conda install:

               platform : win-64
          conda version : 4.2.16
       conda is private : False
      conda-env version : 4.2.16
    conda-build version : 2.1.1
         python version : 3.5.1.final.0
       requests version : 2.12.4
       root environment : C:\Users\laboleb\Anaconda3  (writable)
    default environment : C:\Users\laboleb\Anaconda3
       envs directories : C:\Users\laboleb\Anaconda3\envs
          package cache : C:\Users\laboleb\Anaconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
                          https://conda.anaconda.org/soft-matter/win-64
                          https://conda.anaconda.org/soft-matter/noarch
            config file : C:\Users\laboleb\.condarc
           offline mode : False
</details>

# Not working

<details>
<summary>conda list</summary>
# packages in environment at C:\Users\laboleb\Anaconda3:
#
_nb_ext_conf              0.2.0                    py35_0  
alabaster                 0.7.8                    py35_0  
anaconda                  4.1.0               np111py35_0  
anaconda-client           1.4.0                    py35_0  
anaconda-navigator        1.2.1                    py35_0  
argcomplete               1.0.0                    py35_1  
astropy                   1.2.1               np111py35_0  
babel                     2.3.3                    py35_0  
backports                 1.0                      py35_0  
beautifulsoup4            4.4.1                    py35_0  
bitarray                  0.8.1                    py35_1  
bokeh                     0.11.1                   py35_0  
boto                      2.40.0                   py35_0  
bottleneck                1.0.0               np111py35_1  
bzip2                     1.0.6                    vc14_3  [vc14]
cffi                      1.6.0                    py35_0  
chardet                   2.3.0                    py35_0  
chest                     0.2.3                    py35_0  
click                     6.6                      py35_0  
cloudpickle               0.2.1                    py35_0  
clyent                    1.2.2                    py35_0  
colorama                  0.3.7                    py35_0  
comtypes                  1.1.2                    py35_0  
conda                     4.3.8                    py35_0  
conda-build               2.1.1                    py35_0  
conda-env                 2.6.0                         0  
conda-verify              2.0.0                    py35_0  
configobj                 5.0.6                    py35_0  
console_shortcut          0.1.1                    py35_1  
contextlib2               0.5.3                    py35_0  
cryptography              1.4                      py35_0  
curl                      7.49.0                   vc14_0  [vc14]
cycler                    0.10.0                   py35_0  
cython                    0.24                     py35_0  
cytoolz                   0.8.0                    py35_0  
dask                      0.10.0                   py35_0  
datashape                 0.5.2                    py35_0  
decorator                 4.0.10                   py35_0  
dill                      0.2.5                    py35_0  
docutils                  0.12                     py35_2  
dynd-python               0.7.2                    py35_0  
entrypoints               0.2.2                    py35_0  
et_xmlfile                1.0.1                    py35_0  
fastcache                 1.0.2                    py35_1  
filelock                  2.0.7                    py35_0  
flask                     0.11.1                   py35_0  
flask-cors                2.1.2                    py35_0  
freetype                  2.5.5                    vc14_1  [vc14]
get_terminal_size         1.0.0                    py35_0  
gevent                    1.1.1                    py35_0  
greenlet                  0.4.10                   py35_0  
h5py                      2.6.0               np111py35_0  
hdf5                      1.8.15.1                 vc14_4  [vc14]
heapdict                  1.0.0                    py35_1  
idna                      2.1                      py35_0  
imagesize                 0.7.1                    py35_0  
ipykernel                 4.3.1                    py35_0  
ipython                   4.2.0                    py35_0  
ipython_genutils          0.1.0                    py35_0  
ipywidgets                4.1.1                    py35_0  
itsdangerous              0.24                     py35_0  
jdcal                     1.2                      py35_1  
jedi                      0.9.0                    py35_1  
jinja2                    2.8                      py35_1  
jpeg                      8d                       vc14_0  [vc14]
jsonschema                2.5.1                    py35_0  
jupyter                   1.0.0                    py35_3  
jupyter_client            4.3.0                    py35_0  
jupyter_console           4.1.1                    py35_0  
jupyter_core              4.1.0                    py35_0  
libdynd                   0.7.2                         0  
libpng                    1.6.22                   vc14_0  [vc14]
libtiff                   4.0.6                    vc14_2  [vc14]
llvmlite                  0.11.0                   py35_0  
locket                    0.2.0                    py35_1  
lxml                      3.6.0                    py35_0  
markupsafe                0.23                     py35_2  
matplotlib                1.5.1               np111py35_0  
menuinst                  1.4.1                    py35_0  
mistune                   0.7.2                    py35_0  
mkl                       11.3.3                        1  
mkl-service               1.1.2                    py35_2  
mpmath                    0.19                     py35_1  
multipledispatch          0.4.8                    py35_0  
nb_anacondacloud          1.1.0                    py35_0  
nb_conda                  1.1.0                    py35_0  
nb_conda_kernels          1.0.3                    py35_0  
nbconvert                 4.2.0                    py35_0  
nbformat                  4.0.1                    py35_0  
nbpresent                 3.0.2                    py35_0  
networkx                  1.11                     py35_0  
nltk                      3.2.1                    py35_0  
nose                      1.3.7                    py35_1  
notebook                  4.2.1                    py35_0  
numba                     0.26.0              np111py35_0  
numexpr                   2.6.0               np111py35_0  
numpy                     1.11.0                   py35_2  
odo                       0.5.0                    py35_0  
openpyxl                  2.3.2                    py35_0  
openssl                   1.0.2h                   vc14_0  [vc14]
pandas                    0.18.1              np111py35_0  
partd                     0.3.4                    py35_0  
path.py                   8.2.1                    py35_0  
pathlib2                  2.1.0                    py35_0  
patsy                     0.4.1                    py35_0  
pep8                      1.7.0                    py35_0  
pickleshare               0.7.2                    py35_0  
pillow                    3.2.0                    py35_1  
pip                       8.1.2                    py35_0  
pkginfo                   1.4.1                    py35_0  
ply                       3.8                      py35_0  
psutil                    4.3.0                    py35_0  
py                        1.4.31                   py35_0  
pyasn1                    0.1.9                    py35_0  
pycosat                   0.6.1                    py35_1  
pycparser                 2.14                     py35_1  
pycrypto                  2.6.1                    py35_4  
pycurl                    7.43.0                   py35_0  
pyflakes                  1.2.3                    py35_0  
pygments                  2.1.3                    py35_0  
pyopenssl                 16.2.0                   py35_0  
pyparsing                 2.1.4                    py35_0  
pyqt                      4.11.4                   py35_6  
pyreadline                2.1                      py35_0  
pytables                  3.2.2               np111py35_4  
pytest                    2.9.2                    py35_0  
python                    3.5.1                         5  
python-dateutil           2.5.3                    py35_0  
pytz                      2016.4                   py35_0  
pywin32                   220                      py35_1  
pyyaml                    3.11                     py35_4  
pyzmq                     15.2.0                   py35_0  
qt                        4.8.7                    vc14_8  [vc14]
qtconsole                 4.2.1                    py35_0  
qtpy                      1.0.2                    py35_0  
requests                  2.12.4                   py35_0  
rope                      0.9.4                    py35_1  
ruamel_yaml               0.11.14                  py35_0  
scikit-image              0.12.3              np111py35_1  
scikit-learn              0.17.1              np111py35_1  
scipy                     0.17.1              np111py35_1  
setuptools                23.0.0                   py35_0  
simplegeneric             0.8.1                    py35_1  
singledispatch            3.4.0.3                  py35_0  
sip                       4.16.9                   py35_2  
six                       1.10.0                   py35_0  
snowballstemmer           1.2.1                    py35_0  
sockjs-tornado            1.0.3                    py35_0  
sphinx                    1.4.1                    py35_0  
sphinx_rtd_theme          0.1.9                    py35_0  
spyder                    2.3.9                    py35_0  
sqlalchemy                1.0.13                   py35_0  
statsmodels               0.6.1               np111py35_1  
sympy                     1.0                      py35_0  
tk                        8.5.18                   vc14_0  [vc14]
toolz                     0.8.0                    py35_0  
tornado                   4.3                      py35_1  
traitlets                 4.2.1                    py35_0  
unicodecsv                0.14.1                   py35_0  
vs2015_runtime            14.0.25123                    0  
werkzeug                  0.11.10                  py35_0  
wheel                     0.29.0                   py35_0  
xlrd                      1.0.0                    py35_0  
xlsxwriter                0.9.2                    py35_0  
xlwings                   0.7.2                    py35_0  
xlwt                      1.1.2                    py35_0  
zlib                      1.2.8                    vc14_3  [vc14]
</details>

<details>
<summary>conda info</summary>
Current conda install:

               platform : win-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : 2.1.1
         python version : 3.5.1.final.0
       requests version : 2.12.4
       root environment : C:\Users\laboleb\Anaconda3  (writable)
    default environment : C:\Users\laboleb\Anaconda3
       envs directories : C:\Users\laboleb\Anaconda3\envs
          package cache : C:\Users\laboleb\Anaconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
                          https://conda.anaconda.org/soft-matter/win-64
                          https://conda.anaconda.org/soft-matter/noarch
            config file : C:\Users\laboleb\.condarc
           offline mode : False
             user-agent : conda/4.3.8 requests/2.12.4 CPython/3.5.1 Windows/2012ServerR2 Windows/6.3.9600
</details>

## Diff between conda list output of the two environments

```
$ diff working.txt notworking.txt
26c26
< conda                     4.2.16                   py35_0
---
> conda                     4.3.8                    py35_0
```

**Comment by bkreider** [comment]
@kalefranz If it is related to requests, I had submitted an issue that the conda package for requests doesn't capture the "optional" dependencies' version requirements.  True cryptography is an optional requests dependency, but if you do have cryptography installed it must meet the proper version requirement or requests breaks.

```
extras_require={
        'security': ['pyOpenSSL>=0.14', 'cryptography>=1.3.4', 'idna>=2.0.0'],
        'socks': ['PySocks>=1.5.6, !=1.5.7'],
    },
```

**Comment by kalefranz** [comment]
repodata for conda 4.2.16

    "conda-4.2.16-py35_0.tar.bz2": {
      "build": "py35_0",
      "build_number": 0,
      "date": "2017-01-20",
      "depends": [
        "conda-env >=2.6",
        "menuinst",
        "pycosat >=0.6.1",
        "pyopenssl >=16.2",
        "python 3.5*",
        "requests >=2.12.4",
        "ruamel_yaml >=0.11.14"
      ],
      "license": "BSD",
      "md5": "65ed12f71ab965fe21c7dd77634bcce2",
      "name": "conda",
      "size": 472834,
      "version": "4.2.16"
    },

repodata for conda 4.3.8

    "conda-4.3.8-py35_0.tar.bz2": {
      "build": "py35_0",
      "build_number": 0,
      "date": "2017-01-23",
      "depends": [
        "conda-env >=2.6",
        "menuinst",
        "pycosat >=0.6.1",
        "pyopenssl >=16.2",
        "python 3.5*",
        "requests >=2.12.4",
        "ruamel_yaml >=0.11.14"
      ],
      "license": "BSD",
      "md5": "6e7b84a9c7830d5e4ad903f4a46c9fe8",
      "name": "conda",
      "size": 545934,
      "version": "4.3.8"
    },

@bkreider as of something like conda 4.2.14, we've been including `pyopenssl` as a hard dependency, exactly for this reason.  `cryptography` is a hard dependency of our `pyopenssl` package, so we're bringing that in as well.

    "pyopenssl-16.2.0-py35_0.tar.bz2": {
      "build": "py35_0",
      "build_number": 0,
      "date": "2016-11-09",
      "depends": [
        "cryptography >=1.3.4",
        "python 3.5*",
        "six >=1.5.2"
      ],
      "license": "Apache Software License",
      "license_family": "Apache",
      "md5": "f74e979256c4ce6bb3edc86198d18941",
      "name": "pyopenssl",
      "requires": [],
      "size": 71995,
      "version": "16.2.0"
    },

Now maybe I'm seeing a problem...  I'm pretty sure requests is *wrong* with

    'security': ['pyOpenSSL>=0.14', 'cryptography>=1.3.4', 'idna>=2.0.0'],

That's why we bumped `pyopenssl` all the way up to `16.2`.  But pyopenssl 16.2 might need to have it's cryptography dependency bumped beyond 1.3.4.

@kmdouglass Can you experiment with different versions of cryptography beyond 1.4?

I see the latest is cryptography 1.7.1.  If that fixes the problem, see if the problem comes back with 1.6 and 1.5.3.

**Comment by kmdouglass** [comment]
Thanks @kalefranz 

Updating to conda 4.3.8 *and* cryptography 1.7.1, then trying to install my package produces the following verification error for each individual module in my package (I'm only showing the error for one of the modules for brevity):

```
CondaVerificationError: The package for bstore located at C:\Users\laboleb\Anaco
nda3\pkgs\bstore-1.1.1-py35_0
appears to be corrupted. The path 'lib/python3.5/site-packages/bstore/readers.py'
specified in the package manifest cannot be found.
```

I'm also still getting the `@ffi.def_extern()` error.

If I downgrade back to conda 4.2.16 works and keep cryptography 1.7.1, the package successfully installs.

**Comment by kalefranz** [comment]
The root cause of the `CondaVerificationError` has been fixed in later 4.3 patch releases.  `conda clean --packages --tarballs` usually resolves the problem if you've fallen into a place where conda can't fix it itself automatically.

**Comment by ankitshah009** [comment]
Above solution works always for - conda clean --packages --tarballs. 
My path had issues with incomplete download due to an internet disconnection and resulted in a corrupted package. The above solution helped solve the problems faced.

**Comment by yashkant** [comment]
> The root cause of the CondaVerificationError has been fixed in later 4.3 patch releases. conda clean --packages --tarballs usually resolves the problem if you've fallen into a place where conda can't fix it itself automatically

Is there a way to clean only selected tarballs which might be corrupted?
